### PR TITLE
feat(site): show license utilization in general settings

### PR DESCRIPTION
**Pull request by SasSwart**
This PR is the first iteration towards #15297

We cannot yet show license utilization over time, so we show current license utilization.
This is because we don't track user states over time. We only track the current user state. A graph over time filtering by active users would therefore not account for day to day changes in user state and be inaccurate.
DB schema migrations and related updates will follow that allow us to show license utilization over time.

image

### CODE REVIEW

File: site/src/pages/DeploymentSettingsPage/GeneralSettingsPage/GeneralSettingsPageView.stories.tsx
Comment (johnstcn) [inline site/src/pages/DeploymentSettingsPage/GeneralSettingsPage/GeneralSettingsPageView.stories.tsx:175]: Can we add a story for exceeding license utilization as well?

File: site/src/pages/DeploymentSettingsPage/GeneralSettingsPage/GeneralSettingsPageView.tsx
Comment (johnstcn) [inline site/src/pages/DeploymentSettingsPage/GeneralSettingsPage/GeneralSettingsPageView.tsx:85]: Would it make sense to give these a similar treatment to `licenseUtilizationPercentage` and check if we can evaluate them based on whether `entitlements` is nil?

File: site/src/pages/DeploymentSettingsPage/GeneralSettingsPage/GeneralSettingsPageView.tsx
Comment (mafredri) [inline site/src/pages/DeploymentSettingsPage/GeneralSettingsPage/GeneralSettingsPageView.tsx:45]: Suggestion: Instead of just doing the percentage here, we could prepare the data format in a more convenient way to use in the template:

```typescript
	const licenseLimit =
		entitlements?.features?.user_limit?.actual &&
		entitlements?.features?.user_limit?.limit
			? {
				valid: true,
				actual: entitlements.features.user_limit.actual,
				limit: entitlements.features.user_limit.limit
			} : {
				valid: false,
				actual: 0,
				limit: 0,
			};
```

Might not need to even init the actual/limit for the zero case but added it just in case.

**Review by mafredri — COMMENTED** [review]
Just a minor suggestion, otherwise LGTM (deferring approval for fe).

**Review by johnstcn — APPROVED** [review]
Approving to unblock.

**Review by BrunoQuaresma — APPROVED** [review]
It looks good to me. Thanks for digging into the problem and finding the right solution.

One minor thing, that we can do after merge, is to improve the design a bit. cc.: @chrifro

File: site/src/pages/DeploymentSettingsPage/GeneralSettingsPage/GeneralSettingsPageView.tsx
Comment (aslilac) [inline site/src/pages/DeploymentSettingsPage/GeneralSettingsPage/GeneralSettingsPageView.tsx:68]:
```suggestion
							value={Math.min(licenseUtilizationPercentage * 100, 100)}
```

as is, if the percentage was 105% we would show it as 5%. I think this is what you meant to do. :)

File: site/src/pages/DeploymentSettingsPage/GeneralSettingsPage/GeneralSettingsPageView.stories.tsx
Comment (aslilac) [inline site/src/pages/DeploymentSettingsPage/GeneralSettingsPage/GeneralSettingsPageView.stories.tsx:45]:
```suggestion
```

`undefined` is JavaScript's equivalent to a zero value, this doesn't do anything

**Review by aslilac — APPROVED** [review]
nice!

**Pull request merged**
